L1Trigger/L1TGlobal/src/AXOL1TLCondition.cc

```diff
@@ -0,0 +1,254 @@
+#include "L1Trigger/L1TGlobal/interface/CaloCondition.h"
+#include "L1Trigger/L1TGlobal/interface/EnergySumCondition.h"
+#include "L1Trigger/L1TGlobal/interface/MuonTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/CaloTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/EnergySumTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/GlobalScales.h"
+
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+
+#include "L1Trigger/L1TGlobal/interface/GlobalBoard.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/MessageLogger/interface/MessageDrop.h"
+
+// constructors
+//     default
+l1t::AXOL1TLCondition::AXOL1TLCondition() : ConditionEvaluation() {
+  // empty
+}
+
+//     from base template condition (from event setup usually)
+l1t::AXOL1TLCondition::AXOL1TLCondition(const GlobalCondition* axol1tlTemplate, const GlobalBoard* ptrGTB)
+    : ConditionEvaluation(),
+      m_gtAXOL1TLTemplate(static_cast<const AXOL1TLTemplate*>(axol1tlTemplate)),
+      m_gtGTB(ptrGTB) {}
+
+// copy constructor
+void l1t::AXOL1TLCondition::copy(const l1t::AXOL1TLCondition& cp) {
+  m_gtAXOL1TLTemplate = cp.gtAXOL1TLTemplate();
+  m_gtGTB = cp.gtGTB();
+
+  m_condMaxNumberObjects = cp.condMaxNumberObjects();
+  m_condLastResult = cp.condLastResult();
+  m_combinationsInCond = cp.getCombinationsInCond();
+
+  m_verbosity = cp.m_verbosity;
+}
+
+l1t::AXOL1TLCondition::AXOL1TLCondition(const l1t::AXOL1TLCondition& cp) : ConditionEvaluation() { copy(cp); }
+
+// destructor
+l1t::AXOL1TLCondition::~AXOL1TLCondition() {
+  // empty
+}
+
+// equal operator
+l1t::AXOL1TLCondition& l1t::AXOL1TLCondition::operator=(const l1t::AXOL1TLCondition& cp) {
+  copy(cp);
+  return *this;
+}
+
+// methods
+void l1t::AXOL1TLCondition::setGtAXOL1TLTemplate(const AXOL1TLTemplate* caloTempl) { m_gtAXOL1TLTemplate = caloTempl; }
+
+///   set the pointer to uGT GlobalBoard
+void l1t::AXOL1TLCondition::setuGtB(const GlobalBoard* ptrGTB) { m_gtGTB = ptrGTB; }
+
+const bool l1t::AXOL1TLCondition::evaluateCondition(const int bxEval) const {
+  bool condResult = false;
+  int useBx = bxEval + m_gtAXOL1TLTemplate->condRelativeBx();
+  cout << "Considering BX " << useBx << std::endl;
+
+  //HLS4ML stuff
+  std::string AXOL1TLmodelversion = "GTADModel_v1";
+  cout << "loading model... " << AXOL1TLmodelversion << std::endl;
+  hls4mlEmulator::ModelLoader loader(AXOL1TLmodelversion);
+  std::shared_ptr<hls4mlEmulator::Model> model;
+  model = loader.load_model();
+  cout << "model loaded! " << std::endl;
+
+  // //pointers to objects
+  const BXVector<const l1t::Muon*>* candMuVec = m_gtGTB->getCandL1Mu();
+  const BXVector<const l1t::L1Candidate*>* candJetVec = m_gtGTB->getCandL1Jet();
+  const BXVector<const l1t::L1Candidate*>* candEGVec = m_gtGTB->getCandL1EG();
+  const BXVector<const l1t::EtSum*>* candEtSumVec = m_gtGTB->getCandL1EtSum();
+
+  const int NMuons = 4;
+  const int NJets = 10;
+  const int NEgammas = 4;
+  const int NEtSums = 1;
```

Comment:
This is probably fine, but I would consider if AXOL1TL will at least _consider_ the idea of changing or adding inputs. A less hard-coded solution may be preferable if that's on the table.

Reply:
for now as far as I know this is unlikely to change, so I think we should leave these hardcoded for now. Happy to hear other opinions though

Reply:
Fine for now.